### PR TITLE
feat(ledger): Updates after Ledger review

### PR DESCRIPTION
**Pull request by ixje**

### CODE REVIEW

File: packages/neon-ledger/src/ErrorCode.ts [hunk 1]
```diff
@@ -36,7 +39,7 @@ export interface TransportStatusError extends Error {
  * @returns error with modified message if found.
  */
 export function evalTransportError(err: Error): Error {
-  const transportErr = err as TransportStatusError;
+  const transportErr = cloneDeep(err) as TransportStatusError;
```
[the remark this answers is not in the document]
Comment (ixje) [reply packages/neon-ledger/src/ErrorCode.ts:42]: We had some functional issues on the neon-wallet side after trying to switch to the `evalTransportError` function from `neon-js` as opposed to their baked in version (which is pretty much a copy of neon-js's but has this additional clone in the body). This looked like the safest and fastest way to mirror the working behaviour without having to look into how javascript deals with copying object (by reference?) and possible recursive objects. Maybe the `Error` object is really simple, but there is a lot of time pressure on the Ledger review side that I did not want to spend my time figuring it out and instead use it to deal with their other requests. Not the prettiest answer ¯\_(ツ)_/¯

**Comment by ixje** [comment]
Wait with merging. I need to add more new error codes

File: packages/neon-ledger/src/ErrorCode.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (snowypowers) [reply packages/neon-ledger/src/ErrorCode.ts:43]: sure lets roll with this for now.

**Comment by snowypowers** [comment]
any updates?

**Comment by ixje** [comment]
I'm currently on holiday. The issue is still valid. There is a "but".
Ledger has basically ignored us for 2 months now by saying "we don't have
time" to review the ledger app updates we PR'ed for.

**Comment by snowypowers** [comment]
another ping, seems like ledger got the review through?

**Comment by ixje** [comment]
Yes! I’ll have a look at it today to make sure all fields still match

**Comment by ixje** [comment]
All good now

File: packages/neon-ledger/package.json [hunk 2]
```diff
@@ -37,7 +37,9 @@
   },
   "dependencies": {
     "@ledgerhq/hw-transport": "5.51.1",
-    "@types/ledgerhq__hw-transport": "4.21.3"
+    "@types/ledgerhq__hw-transport": "4.21.3",
+    "lodash-es": "4.17.15",
```
Comment (snowypowers) [inline packages/neon-ledger/package.json:41]: is this an accidental addition? I dont see it used anywhere

Comment (ixje) [reply packages/neon-ledger/package.json:41]: `cloneDeep` is import from `lodash-es` and used in `ErrorCode.ts`, i'll ping you there

**Pull request merged**